tests/common/mod.rs

```diff
@@ -0,0 +1,130 @@
+use labrinth::clickhouse;
+use labrinth::{
+    file_hosting, queue,
+    util::env::{parse_strings_from_var, parse_var},
+    LabrinthConfig,
+};
+use std::sync::Arc;
+
+use self::database::TemporaryDatabase;
+
+pub mod actix;
+pub mod database;
+pub mod environment;
+
+// Testing equivalent to 'setup' function, producing a LabrinthConfig
+// If making a test, you should probably use environment::TestEnvironment::new() (which calls this)
+pub async fn setup(db: &TemporaryDatabase) -> LabrinthConfig {
+    println!("Setting up labrinth config");
+
+    dotenvy::dotenv().ok();
+
+    if check_test_vars() {
+        println!("Some environment variables are missing!");
+    }
+
+    let pool = db.pool.clone();
+    let redis_pool = db.redis_pool.clone();
+    let file_host: Arc<dyn file_hosting::FileHost + Send + Sync> =
+        Arc::new(file_hosting::MockHost::new());
+    let mut clickhouse = clickhouse::init_client().await.unwrap();
+
+    let maxmind_reader = Arc::new(queue::maxmind::MaxMindIndexer::new().await.unwrap());
+
+    labrinth::app_setup(
+        pool.clone(),
+        redis_pool.clone(),
+        &mut clickhouse,
+        file_host.clone(),
+        maxmind_reader.clone(),
+    )
+}
+
+// This is so that env vars not used immediately don't panic at runtime
+// Currently, these are the same as main.rs ones.
+// TODO: go through after all tests are created and remove any that are not used
+// Low priority as .env file should include all of these anyway
+fn check_test_vars() -> bool {
```

Comment:
doesn't labrinth's init function already call this (which is called from the tests). if not, couldn't you just reuse the labrinth function? also might be annoying to add env variables here as well when you add a new one

Reply:
Yeah, I thought there may be a case where they may be different (ie: ability to run tests without env vars). But I guess it doesn't matter so much- .env exists